File: recursive/ack.ml
let rec ack x y = match (x,y) with
    | (0, _) -> y + 1
    | (_ , 0) -> ack (x - 1) 1
    | (_, _) -> ack (x - 1) (ack x (y - 1));;

for it = 1 to 1000 do let _ = (ack 3 6) in () done

File: recursive/ack.py
def ack(m, n):
	if m == 0:
		return n + 1
	elif n == 0:
		return ack(m-1, 1)
	else:
		return ack(m-1, ack(m,n-1))

if __name__ == "__main__":
    for i in range(1000):
        ack(3, 6)